pyomo/core/expr/logical_expr.py

```diff
@@ -198,19 +168,16 @@ def __getstate__(self):
             state[i] = getattr(self, i)
         return state
 
-    def __nonzero__(self):
-        if _using_chained_inequality and not self.is_constant():
-            deprecation_warning("Chained inequalities are deprecated. "
-                                "Use the inequality() function to "
-                                "express ranged inequality expressions.")     # Remove in Pyomo 6.0
-            _chainedInequality.call_info = traceback.extract_stack(limit=2)[-2]
-            _chainedInequality.prev = self
-            return True
-            #return bool(self())                # This is needed to apply simple evaluation of inequalities
-
-        return bool(self())
-
-    __bool__ = __nonzero__
+    def __bool__(self):
+        raise PyomoException('Cannot convert non-constant expression '
+                             'to bool. This error usually arises for '
+                             'one of two reasons. One possibility is '
+                             'using an expression in a boolean context '
```

Comment:
Error message states that there are two common reasons why this exception might be raised but then only describes one.

Reply:
Fixed. Thanks!